includes/components/class-admin.php

```diff
@@ -1867,4 +1904,30 @@ function( $config ) {
 			]
 		);
 	}
+
+	/**
+	 * Update dismissed HivePress pointers.
+	 */
+	public function update_dismiss_pointers() {
+
+		// Get pointer.
+		$pointer = sanitize_text_field( hp\get_array_value( $_POST, 'pointer' ) );
+
+		if ( ! current_user_can( 'edit_others_posts' ) || ! $pointer ) {
+			return;
+		}
+
+		// Get dismissed HivePress pointers.
+		$dismissed = (array) get_user_meta( get_current_user_id(), 'hp_dismissed_pointers', true );
+
+		if ( in_array( $pointer, $dismissed ) ) {
+			return;
+		}
+
+		// Add dismissed HivePress pointers.
+		$dismissed[] = $pointer;
+
+		// Update dismissed HivePress pointers.
+		update_user_meta( get_current_user_id(), 'hp_dismissed_pointers', $dismissed );
```

Comment:
it may be a bit easier if you use array_unique(array_merge($dismissed... without the above checks, but saving extra strings is not safe so should be available to admins only